README.md

```diff
@@ -38,6 +38,17 @@ We also support to run each benchmark once (with only one fork and one iteration
 mvn test -P test
 ```
 
+## Prerequisites
+
+The recent addition of OpenSSL-based benchmarks require one of two modes to be active:
+- dynamically-linked OpenSSL (default): this uses a dependency with native wrappers that are linked dynamically to your system's libraries. Depending on the installed OpenSSL version and OS, this may fail and you should try the next option:
+- statically-linked OpenSSL: this can be activated by `mvn -Dinclude-netty-tcnative-static` but requires `flink-shaded-netty-tcnative-static` in the version from `pom.xml` which can be generated from inside a corresponding flink-shaded source via:
+```
+mvn clean install -Pinclude-netty-tcnative-static -pl flink-shaded-netty-tcnative-static
+```
+
+If both options are not working, OpenSSL benchmarks will fail but that should not influence any other benchmarks.
```

Comment:
Maybe instead of failing, default action should be not execute them? However I'm not sure how to achieve that with `@Params` from JMX :/

What's the error message that's going to be displayed? Maybe we could wrap it if it's not clear enough?

Do you know what will happen on the benchmarking machine?

Reply:
Depending on the exact error, the spawned JVM process may just die and we can't do anything to prevent that. Luckily, JMH just continues in that case, so the final results will just not include the failed benchmarks

Reply:
As it is now, those benchmarks would fail on the Hetzner worker? Do you think it's worth spending time to try make them work? Would it involve some extra maintenance cost when upgrading Flink/netty?

Reply:
Good question, Debian- and RedHat-based distros should just work with our dynamically-linked OpenSSL bindings once [dependencies](https://netty.io/wiki/forked-tomcat-native.html#prerequisites-for-dynamically-linked-netty-tcnative) are installed: usually only libapr1 may be missing
If that is not usable, building the statically-linked netty-native is quite simple (as mentioned) since this basically only downloads appropriate artifacts. That should always work and here, we also do not need to look after licensing issues
Maintenance-wise, there could be overhead due to upgrades in the dynamically-linked version. Statically-linked bindings (when generated automatically for the appropriate Flink version) should come at no additional overhead.

Reply:
Ok, thanks for the explanation @NicoK. I would leave it up to you if it's worth doing or not. I guess there is not that much of a value for us to benchmark OpenSSL, except of having some baseline numbers for recommendations Flink users, so I would be fine with NOT benchmarking it.

Reply:
I would actually like to keep these benchmarks as they may also show differences when we change configuration defaults that may have different effects on JDK SSL vs. OpenSSL.
It's just a single benchmark anyway - but if you don't want it in the per-commit runs, we could also move them to the ones you'd run manually

Reply:
I meant that I wouldn't spend too much time in trying to make them work on Hetzner, but if you would like to try to do it, I'm +1 for it. The code for the benchmarks we can keep it has value, even if we are not using it, as long as it's not causing as extra maintenance :)

In other words, LGTM for the code changes, and it's up to you if you want to make sure OpenSSL is executing properly on Hetzner machine.